bin/main

```diff
@@ -1,128 +1,79 @@
 
 instructions = "1.The game is played on a 3x3 grid.\n
 2. One player is X, the other is O.\n
 3. Player take turns to input a no. between (1-9) to mark one of the empty squares.\n
 4. The first player to get 3 marks in a row (up, down, across, or diagonally) is the winner.\n
 5. If none of the players win and all 9 squares are filled the game is a draw"
 
-p1_arr = []
-p2_arr = []
 game_ctr = 0
+p1 = Player.new
+p2 = Player.new
 
 puts "xoxoxoxoxoxo Welcome to the Game of Tic-Tact-Toe  xoxoxoxoxoxo\n\n"
 puts "               Game Instructions\n\n"
+
 puts instructions + "\n\n"
+
 puts "\n3x3 Grid\n"
-(1..9).each do |i|
-  print " #{i} "
-  print '|' unless (i % 3).zero?
-  puts "\n" if (i % 3).zero?
-end
+Board.print_board('', '', [], [])
+
 puts "\nWhat the grid may look like when there is a winner\n"
-(1..9).each do |i|
-  print ' X ' if i <= 3
-  print " #{i} " if i > 3
-  print '|' unless (i % 3).zero?
-  puts "\n" if (i % 3).zero?
-end
-puts "\nWhat the grid may look like when there is a draw\n"
+Board.print_board('X', '', [1, 2, 3], [])
 
-print "  X  O  X\n"
-print "  O  X  O\n"
-print "  O  X  O\n"
+puts "\nWhat the grid may look like when there is a draw\n"
+Board.print_board('X', 'O', [1, 3, 5, 8], [2, 4, 6, 7, 9])
 
 print "\nWant to play the game (y/n)? = "
 want_to_play = gets.chomp.capitalize
 
 while want_to_play == 'Y'
-  p1_arr = []
-  p2_arr = []
+
+  p1.moves = []
+  p2.moves = []
   game_ctr = 0
   flag = 0
+
+  print "\nPlayer 1 write your name = "
+  p1.name = gets.chomp.to_s
+
   while flag.zero?
-    print "\nPlayer 1 choose your symbol (O / X) = "
-    p1_sym = gets.chomp.capitalize
-    flag = 1 if %w[O X].include?(p1_sym)
+    print "\n#{p1.name} choose your symbol (O / X) = "
+    p1.symbol = gets.chomp.capitalize
+    flag = 1 if %w[O X].include?(p1.symbol)
     puts "\nInvalid symbol. Choose again\n" if flag.zero?
   end
 
-  p2_sym = p1_sym == 'X' ? 'O' : 'X'
-  puts "\nPlayer 2 your symbol is = " + p2_sym
+  print "\nPlayer 2 write your name = "
+  p2.name = gets.chomp.to_s
 
-  while game_ctr < 5
+  p2.symbol = p1.symbol == 'X' ? 'O' : 'X'
+  puts "\n#{p2.name} your symbol is = " + p2.symbol
 
+  while game_ctr < 5
     flag = 0
     while flag.zero?
-      print "\nPlayer 1 make your move = "
+      print "\n#{p1.name}-#{p1.symbol} make your move = "
       p1_move = gets.chomp
 
-      if validmove(p1_move.to_i, p1_arr, p2_arr)
+      if Game.validmove(p1_move.to_i, p1.moves, p2.moves)
         flag = 1
       else
         puts "\nThat was an Invalid Move. Input move again\n"
       end
 
     end
 
-    p1_arr.push(p1_move.to_i)
+    p1.moves.push(p1_move.to_i)
 
-    board(p1_sym, p2_sym, p1_arr, p2_arr)
+    Board.print_board(p1.symbol, p2.symbol, p1.moves, p2.moves)
 
-    if checkwin(p1_move.to_i, p1_arr)
-      puts "\nPlayer1 CONGRATULATIONS!!! you are the WINNER\n"
+    if Game.checkwin(p1_move.to_i, p1.moves)
+      puts "\n#{p1.name} CONGRATULATIONS!!! you are the WINNER\n"
       break
     end
```

Comment:
[Please fix]
- [ ] DRY rule should be respected. you are doing exactly the same thing for player one and player two instead of creating a method and reuse it. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/dry.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/)) In order to fix this, you can create a method in your main.rb file that accepts the `player`, `game`, and `board` parameters and move all that logic inside of it.